core/files.go

```diff
@@ -0,0 +1,90 @@
+// Copyright © 2020 Solus Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+var (
+	// TrackDir is an externally defined location used to store the TriggerFile
+	TrackDir string
+	// SysDir is an externally defined location wherein system migration files are stored
+	SysDir string
+	// UsrDir is an externally defined location wherein user migration files are stored
+	UsrDir string
+	// TriggerFile is the location of the trigger file, which must be present for migration to occur
+	TriggerFile = TrackDir + "/trigger"
+)
+
+func createDir(path string) error {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		if err := os.Mkdir(path, 0644); err != nil {
+			return err
+		}
+	} else if err != nil {
```

Comment:
If you define `var err error` at the start of the function, you can just `return err` and it will be nil if there is no error.